Lib/subprocess.py

```diff
@@ -1502,7 +1533,9 @@ def _wait(self, timeout):
 
 
         def _readerthread(self, fh, buffer):
-            buffer.append(fh.read())
+            data = fh.read()
```

Comment:
How does one then handle a disparity of rates between how quickly one can read from `fh` and how quickly they can write to `sys.stdout`?  I don't see a simple solution, unless I've misunderstood what you're saying.

Reply:
I don't expect any `tee` like process to handle this. The posix side of things in this code doesn't either, it reads up to 32k when the selector event says the pipe has data and calls the callback on that.  if the callback blocks, the child process can thus also block on IO when one of its pipe buffers fills up.

there's also the issue of threading, if a callback _might_ be called from a different thread rather than in the thread calling Popen.communicate, that needs to be documented.

Reply:
I'm wondering if that's a constraint of the original design that we're just uncovering?  Is that really an issue to fix here, or just something to call out and having been exposed...

Reply:
The original design consumes all pipe data as it comes in and buffers it as it is only available as a whole upon process exit when `communicate()` returns. The reason I bring this up here is that the tee and callback behavior would be inconsistent between Windows and posix such that the a read callback and tee wouldn't be very useful on Windows.

Reply:
I can add code to disable the `tee` functionality on Windows.

Reply:
Do you want me to proceed with that change?

Reply:
@gpshead I don't know how to proceed.  It would seem that the functionality of this feature has additional contractual obligations on Windows that aren't implied anywhere else.  Should I just limit the functionality of the baked-in "tee" functionality to non-Windows (i.e. POSIX) platforms instead?

Reply:
I've disabled the `tee` functionality for Windows.

Reply:
This still leaves Windows behind, it won't make callbacks in real time as data comes in. It isn't about `tee` so much as it's about having a similar user experience and behavior on both platforms. We need to conditionally do bounded read()s in a loop on windows when a callback is set.

From a user point of view "good enough" is likely be to call `fh.readline(2000)` in a loop in this situation. The things someone wants to see data from live are often text based so reading by line, if present, otherwise a limited buffer is roughly the same to non-blocking reads of everything available. something like this:

```
if not callback:
    buffer.append(fh.read())  # the existing logic
else:
    data = None
    while data != b'':
        data = fh.readline(2000)
        callback(self, buffer, data)
```

Reply:
> so much as it's about having a similar user experience and behavior on both platforms

It seems like we're bifurcating the behavior on Windows depending on whether the callback has been installed or not.  Why not just have consistent and predictable behavior?